### PR TITLE
Fix numpy type hints + CI

### LINKED ISSUE #558: Scheduled CI failes on mypy checks for numpy arrays (py3.7 and py3.8)

### PR DESCRIPTION
**Pull request by asnyv**
---

### Contributor checklist

- [X] :tada: This PR closes #558.
- [X] :scroll: I have broken down my PR into the following tasks:
   - [X] Unrelated fix in GitHub Actions workflow (env cleaning section), ref: https://github.com/actions/virtual-environments/issues/1057
   - [X] Fix numpy type hints
- [ ] :robot: I have added tests, or extended existing tests, to cover any new features or bugs fixed in this PR.
- [ ] :book: I have considered adding a new entry in `CHANGELOG.md`, and added it if should be communicated there.

### CODE REVIEW

File: .github/workflows/subsurface.yml [hunk 1]
```diff
@@ -30,7 +30,7 @@ jobs:
       run: |
         # https://github.com/actions/virtual-environments/issues/751
         # https://github.com/actions/virtual-environments/issues/709
-        sudo apt-get purge p7zip* yarn ruby-full zulu-*-azure-jdk* ghc* php7*
+        sudo apt-get purge p7zip* yarn ruby-full adoptopenjdk* ghc* php7*
```
Comment (asnyv) [inline .github/workflows/subsurface.yml:33]: Unrelated, see:
https://github.com/actions/virtual-environments/issues/1057

File: webviz_subsurface/plugins/_well_cross_section_fmu.py [hunk 2]
```diff
@@ -532,9 +532,13 @@ def _render_section(
             layout.update(self.plotly_theme["layout"])
             if "keep_zoom_state" in options:
                 layout["uirevision"] = "keep"
-            fencespec: List[List[float]] = [
-                [coord[0], coord[1]] for coord in xsect.fence
-            ]
+            # [[]] "solves" type hint, but is pretty dirty
+            # and probably wrong...
```
Comment (asnyv) [inline webviz_subsurface/plugins/_well_cross_section_fmu.py:536]: what to do? 🤔

**Comment by asnyv** [comment]
Made a couple of changes to the pvt code to use numpy arrays "all over". Need a bit of testing to make sure it isn't buggy, but seems ok at first glance.

File: webviz_subsurface/plugins/_well_cross_section_fmu.py (hunk 2 above)
Comment (HansKallekleiv) [reply webviz_subsurface/plugins/_well_cross_section_fmu.py:536]: Will not mypy allow Initializing as empty list ?
Tested quickly on 3.6 and 3.8 with no errors.
```python
fencespec: List[List[float]] = []
if xsect.fence is not None:
    fencespec = [[coord[0], coord[1]] for coord in xsect.fence]
```

**Review by HansKallekleiv — CHANGES_REQUESTED** [review]

File: webviz_subsurface/plugins/_well_cross_section_fmu.py (hunk 2 above)
Comment (asnyv) [reply webviz_subsurface/plugins/_well_cross_section_fmu.py:536]: that is likely, probably didn't try 😉

**Review by HansKallekleiv — APPROVED** [review]
LGTM 👍

**Pull request merged**